src/shared/machine-credential.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "creds-util.h"
+#include "escape.h"
+#include "extract-word.h"
+#include "fileio.h"
+#include "macro.h"
+#include "memory-util.h"
+#include "machine-credential.h"
+#include "path-util.h"
+#include "string-util-fundamental.h"
+
+static void machine_credential_free(MachineCredential *cred) {
```

Comment:
We usually call this `_done`, since it's not `free`-ing the object itself.

Reply:
I will rename to machine_credential_done.

Reply:
Fixed in f8c2832218.